redash/models/__init__.py

```diff
@@ -67,488 +67,6 @@ def get(self, query_id):
-        else:
-            d['auth_type'] = 'password'
-
-        if with_api_key:
-            d['api_key'] = self.api_key
-
-        return d
-
-    def is_api_user(self):
-        return False
-
-    @property
-    def profile_image_url(self):
-        if self._profile_image_url is not None:
-            return self._profile_image_url
-
-        email_md5 = hashlib.md5(self.email.lower()).hexdigest()
-        return "https://www.gravatar.com/avatar/{}?s=40&d=identicon".format(email_md5)
-
-    @property
-    def permissions(self):
-        # TODO: this should be cached.
-        return list(itertools.chain(*[g.permissions for g in
-                                      Group.query.filter(Group.id.in_(self.group_ids))]))
-
-    @property
-    def is_invitation_pending(self):
-        return self.password_hash is None
-
-    @classmethod
-    def get_by_email_and_org(cls, email, org):
-        return cls.query.filter(cls.email == email, cls.org == org).one()
-
-    @classmethod
-    def get_by_api_key_and_org(cls, api_key, org):
-        return cls.query.filter(cls.api_key == api_key, cls.org == org).one()
-
-    @classmethod
-    def all(cls, org):
-        return cls.query.filter(cls.org == org).filter(cls.disabled_at == None)
-
-    @classmethod
-    def search(cls, base_query, term):
-        term = u'%{}%'.format(term)
-        search_filter = or_(cls.name.ilike(term), cls.email.like(term))
-
-        return base_query.filter(search_filter)
-
-    @classmethod
-    def all_disabled(cls, org):
-        return cls.query.filter(cls.org == org).filter(cls.disabled_at != None)
-
-    @classmethod
-    def all_not_disabled(cls, org):
-        return cls.all(org).filter(cls.disabled_at == None)
-
-    @classmethod
-    def find_by_email(cls, email):
-        return cls.query.filter(cls.email == email)
-
-    def __str__(self):
-        return u'%s (%s)' % (self.name, self.email)
-
-    def hash_password(self, password):
-        self.password_hash = pwd_context.encrypt(password)
-
-    def verify_password(self, password):
-        return self.password_hash and pwd_context.verify(password, self.password_hash)
-
-    def update_group_assignments(self, group_names):
-        groups = Group.find_by_name(self.org, group_names)
-        groups.append(self.org.default_group)
-        self.group_ids = [g.id for g in groups]
-        db.session.add(self)
-        db.session.commit()
-
-    def has_access(self, obj, access_type):
-        return AccessPermission.exists(obj, access_type, grantee=self)
-
-    def get_id(self):
-        identity = hashlib.md5("{0},{1}".format(self.email, self.password_hash)).hexdigest()
-        return u"{0}-{1}".format(self.id, identity)
-
-
-class Configuration(TypeDecorator):
-    impl = db.Text
-
-    def process_bind_param(self, value, dialect):
-        return value.to_json()
-
-    def process_result_value(self, value, dialect):
-        return ConfigurationContainer.from_json(value)
```

Comment:
Didn't I do this already?

Reply:
![image](https://user-images.githubusercontent.com/289488/50769553-f9f7d180-128c-11e9-8127-81df0f2aa829.png)

Reply:
![](https://media.giphy.com/media/4pMX5rJ4PYAEM/giphy.gif)